src/main/java/com/ericsson/ei/rules/RulesHandler.java

```diff
@@ -39,38 +44,91 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 @Component
-@Scope(value="thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
+@Scope(value = "thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
 public class RulesHandler {
     private static final Logger LOGGER = LoggerFactory.getLogger(RulesHandler.class);
 
-    @Value("${rules.path}") private String jsonFilePath;
+    private static final String EI_HOME_DEFAULT_NAME = ".eiffel-intelligence";
+
+    @Value("${rules.path}")
+    private String jsonFilePath;
 
     private JmesPathInterface jmesPathInterface = new JmesPathInterface();
     private static String jsonFileContent;
     private static JsonNode parsedJson;
-    
+    private String eiHomePath;
+
     public RulesHandler() {
         super();
     }
-    
+
     public void setParsedJson(String jsonContent) throws JsonProcessingException, IOException {
         ObjectMapper objectmapper = new ObjectMapper();
         parsedJson = objectmapper.readTree(jsonContent);
     }
-    
-    public String readRuleFileContent(String ruleFilePath) throws IOException {
-        String ruleJsonFileContent;
+
+    public String readRuleFileContent(String path) throws Exception {
+        String ruleJsonFileContent = null;
+        String ruleFilePath = "";
+        if (checkIfPathIsURLAndJSON(path)) {
+            setEIHomePath();
+            ruleFilePath = downloadRuleFileToHomeDirectory(path);
+        } else {
+            ruleFilePath = path;
+        }
+        if (!ruleFilePath.equals("")) {
+            ruleJsonFileContent = readRulesFileContent(ruleFilePath);
+        }
+        if (ruleFilePath.equals("") || ruleJsonFileContent.equals("")) {
+            throw new Exception("Rules content cannot be empty");
+        }
+        return ruleJsonFileContent;
+    }
+
+    public String readRulesFileContent(String ruleFilePath) throws IOException {
+        String ruleJsonFileContent = null;
         InputStream in = this.getClass().getResourceAsStream(ruleFilePath);
```

Comment:
InputStream is usually named is or use the full name

Reply:
Done.